src/bp/nlu-core/training-pipeline.ts

```diff
@@ -205,36 +206,38 @@ const TrainIntentClassifiers = async (
 const TrainContextClassifier = async (input: TrainStep, tools: Tools, progress: progressCB): Promise<string> => {
   debugTraining('Training context classifier')
 
-  const { list_entities, pattern_entities } = input
-  const customEntities = [...list_entities.map(e => e.entityName), ...pattern_entities.map(e => e.name)]
+  const { languageCode, intents, contexts, list_entities, pattern_entities, nluSeed } = input
 
-  const points = _.flatMapDeep(input.contexts, ctx => {
-    return input.intents
+  const rootIntents = contexts.map(ctx => {
+    const utterances = _(intents)
       .filter(intent => intent.contexts.includes(ctx))
-      .map(intent =>
-        intent.utterances.map(utt => ({
-          label: ctx,
-          coordinates: getCtxFeatures(utt, customEntities)
-        }))
-      )
-  }).filter(x => x.coordinates.filter(isNaN).length === 0)
-
-  const classCount = _.uniq(points.map(p => p.label)).length
-  if (points.length === 0 || classCount <= 1) {
-    progress()
-    debugTraining('No context to train')
-    return ''
-  }
-
-  const svm = new tools.mlToolkit.SVM.Trainer()
-
-  const seed = input.nluSeed
-  const model = await svm.train(points, { kernel: 'LINEAR', classifier: 'C_SVC', seed }, p => {
-    progress(_.round(p, 1))
+      .flatMap(intent => intent.utterances)
+      .value()
+
+    return <Intent<Utterance>>{
+      name: ctx,
+      contexts: [],
+      slot_definitions: [],
+      utterances
+    }
   })
 
+  const rootIntentClassifier = new SvmIntentClassifier(tools, getCtxFeatures)
```

Comment:
I'm not sure of what is the idea behind the "root" naming can you explain that ? we can do by voice if necessary

Reply:
Its a context classifier. Root intent is kinda of the new name of contexts. I'm eventually going to get rid of the notion of context, but I havnt done it yet